core/application/bloodDonationWorkflow/BloodDonationService.ts

```diff
@@ -21,42 +26,49 @@ export class BloodDonationService {
     donationAttributes: BloodDonationAttributes,
     bloodDonationRepository: Repository<DonationDTO, DonationFields>,
     model: BloodDonationModel
-  ): Promise<string> {
-    try {
-      await this.checkDailyRequestThrottling(
-        donationAttributes.seekerId,
-        bloodDonationRepository,
-        model
-      )
+  ): Promise<BloodDonationResponseAttributes> {
+    await this.checkDailyRequestThrottling(
+      donationAttributes.seekerId,
+      bloodDonationRepository,
+      model
+    )
 
-      const validationResponse = validateInputWithRules(
-        {
-          bloodQuantity: donationAttributes.bloodQuantity,
-          donationDateTime: donationAttributes.donationDateTime
-        },
-        validationRules
+    const validationResponse = validateInputWithRules(
+      {
+        bloodQuantity: donationAttributes.bloodQuantity,
+        donationDateTime: donationAttributes.donationDateTime
+      },
+      validationRules
+    )
+    if (validationResponse !== null) {
+      throw new BloodDonationOperationError(
+        `Invalid parameters for blood donation request. ${validationResponse}`,
+        GENERIC_CODES.ERROR
```

Comment:
This error code should be 400 not 500.

Reply:
Updated in f98450b71743d5473aa397d4854e7347002cc21c